docs/blocks/precise-rotation.md

```diff
@@ -239,71 +231,110 @@ For head-like rotation, you need to add 2 states to your block:
-// How many 16ths of 360 is the head rotation? - rounded
-t.rotation = math.round(t.positive_head_rot / 22.5);
+<CodeHeader>BP/scripts/main.js</CodeHeader>
 
-// 0 and 16 represent duplicate rotations (0 degrees and 360 degrees), so 0 is returned if the value of `t.rotation` is 16
-return t.rotation != 16 ? t.rotation;
+```js
+import "./shell.js";
 ```
 
-On a single line so that you can put it into JSON:
+<br>
+
+Now, in our `shell.js` file, we need to import the `world` object from [`@minecraft/server`](https://learn.microsoft.com/minecraft/creator/scriptapi/minecraft/server/minecraft-server):
 
-<CodeHeader>minecraft:block > components > minecraft:on_player_placing > condition</CodeHeader>
+<CodeHeader>BP/scripts/shell.js</CodeHeader>
 
-```c
-t.positive_head_rot = q.head_y_rotation(0) + 360 * (q.head_y_rotation(0) != math.abs(q.head_y_rotation(0))); t.rotation = math.round(t.positive_head_rot / 22.5); return t.rotation != 16 ? t.rotation;
+```js
+import { world } from "@minecraft/server";
 ```
 
-## Applying Rotation
+## Rotation Calculation
+
+Rather than manually typing bounds for each `wiki:rotation` value, you can use some division and rounding to return the values desired!
+
+Add the following function to your script:
+
+<CodeHeader>BP/scripts/shell.js</CodeHeader>
+
+```js
+/** @param {number} playerYRotation */
+function getPreciseRotation(playerYRotation) {
+  // Transform player's head Y rotation to a positive
+  if (playerYRotation < 0) playerYRotation += 360;
+  // How many 16ths of 360 is the head rotation? - rounded
+  const rotation = Math.round(playerYRotation / 22.5);
+
+  // 0 and 16 represent duplicate rotations (0 degrees and 360 degrees), so 0 is returned if the value of `rotation` is 16
+  return rotation !== 16 ? rotation : 0;
+}
+```
 
-Time to use this Molang to set the block properties you have added!
+## Setting Rotation
 
-We will update our block properties when the player is placing the block. This means that in our event, we have access to `q.block_face` and `q.head_y_rotation`.
+Time to use this function to set the block state you have added!
 
-Create the following component and event in your block:
+We will update our block state before the block is placed by using [custom components](/blocks/block-events) and, more specifically, the [beforeOnPlayerPlace](/blocks/block-events#before-player-place) hook. This means that, in our event handler, we have access to the player's rotation.
+
+Add the following to your script to register a custom `wiki:shell_rotation` component:
+
+:::tip
+Think of a unique custom component identifier. There can't be duplicate custom components between packs!
+:::
+
+<CodeHeader>BP/scripts/shell.js</CodeHeader>
+
+```js
+world.beforeEvents.worldInitialize.subscribe(({ blockTypeRegistry }) => {
+  blockTypeRegistry.registerCustomComponent("wiki:shell_rotation", {
+    beforeOnPlayerPlace(e) {
```

Comment:
Would much prefer if `e` was renamed to `eventData` or similar to describe what information it actually contains